source/Calamari.Shared/Integration/Scripting/WindowsPowerShell/Bootstrap.ps1

```diff
@@ -395,6 +395,39 @@ function Decrypt-Variables($iv, $Encrypted)
 	return $parameters
 }
 
+function Set-ProxyEnvironmentVariables ([string] $proxyHost, [int] $proxyPort, [string] $proxyUsername, [string] $proxyPassword) {
+    $proxyUri = Get-ProxyUri -proxyHost $proxyHost -proxyPort $proxyPort
+    if (![string]::IsNullOrEmpty($proxyUsername)) {
+		Add-Type -AssemblyName System.Web
+		$proxyUri = "http://$( [System.Web.HttpUtility]::UrlEncode($proxyUsername) ):$( [System.Web.HttpUtility]::UrlEncode($proxyPassword) )@$( $proxyHost ):$( $proxyPort )"
+	}
+	
+	if([string]::IsNullOrEmpty($env:HTTP_PROXY)) {
+		$env:HTTP_PROXY = "$proxyUri"
+	}
+	
+	if([string]::IsNullOrEmpty($env:HTTPS_PROXY)) {
+		$env:HTTPS_PROXY = "$proxyUri"
+	}
+	
+	if([string]::IsNullOrEmpty($env:NO_PROXY)) {
+		$env:NO_PROXY="127.0.0.1,localhost,169.254.169.254"
+	}
+}
+
+function Set-ProxyEnvironmentVariablesFromSystemProxy([string] $proxyUsername, [string] $proxyPassword) {
+	$testUri = New-Object Uri("https://octopus.com")
+	$systemProxy = [System.Net.WebRequest]::GetSystemWebProxy().GetProxy($testUri)
+	if ($systemProxy.Host -ne "octopus.com") {
+		Set-ProxyEnvironmentVariables -proxyHost $systemProxy.Host -proxyPort $systemProxy.Port -proxyUsername $proxyUsername -proxyPassword $proxyPassword
+	}
+}
+
+function Get-ProxyUri ([string] $proxyHost, [int] $proxyPort) {
+	$uri = "http://${proxyHost}:$proxyPort"
```

Comment:
Shouldn't it be `https` for `$env:HTTPS_PROXY` ?

Reply:
This setting is consistent with Octopus Server and Tentacle https://github.com/OctopusDeploy/OctopusShared/blob/4d87e1cd0d6428b40c832523f387b23edda0885f/source/Octopus.Shared/Configuration/ProxyConfigParser.cs#L71